classes/shop/Shop.php

```diff
@@ -529,7 +529,7 @@ public function getBaseURL($auto_secure_mode = false, $add_base_uri = true)
         }
 
         $url = array();
-        $url['protocol'] = $auto_secure_mode && Tools::usingSecureMode() ? 'https://' : 'http://';
+        $url['protocol'] = Tools::getShopProtocol();
```

Comment:
I'm unsure this change is a good idea, as we can have a different domain per protocol. I'd like to keep the match.

Reply:
Fixed.